linode/manager
# feat: [M3-7143] - Add 2FA Status and Last Login to the Account Users Table

## Description 📝

- Add **2FA Status** 📱 and **Last Login** 🔑 to the Account Users Table (http://localhost:3000/account/users) 🎉

## Changes  🔄
- Adds a TFA chip to the Account Users table if a user has TFA enabled
- Adds a "Last Login" column to the Account Users table
- Fixes `<Chip />` `color` prop  (previously it did nothing 😣)
- Adds ordering to the `Email` and `Username` columns
- Updates the api-v4 package to contain new types returned by the API 📦
- Allows OrderBy utils and hooks to have no initial order
  - This allows us to have `http://localhost:3000/account/users` retain the API's default order until the user decides to sort by clicking a column

## Preview 📷

| Before  | After   |
| ------- | ------- |
| ![Screenshot 2023-10-17 at 5 32 43 PM](https://github.com/linode/manager/assets/115251059/1497f944-c9ca-42fa-bd3d-dc64558f7cab) | ![Screenshot 2023-10-17 at 5 32 31 PM](https://github.com/linode/manager/assets/115251059/438e96ca-db7f-470e-be97-5274bc29b97b) |

## How to test 🧪

### Verification steps 
- Go to `http://localhost:3000/account/users`
- Verify you can **add a user**
- Verify you can **delete a user**
- Verify that a user with 2FA enabled has a 2FA chip (Enable 2FA at http://localhost:3000/profile/auth)
- Verify all possible states of the "Last Login" column (you can turn on the MSW to help with this)
- Verify styling of `<Chips />` throughout the application 🎨 (this component had very bad theming and uses weird defaults)

## As an Author I have considered 🤔

*Check all that apply*
